File: go/sort/select_sort.go
package main

import (
	"fmt"
	"go/util"
)

func main() {

	var arr = []int{12, 4, 6, 2, 5, 1, 35, 78, 21, 9}

	select_sort(arr)

	fmt.Println(arr)

}

func select_sort(arr []int) {

	var i int
	for ; i < len(arr); i++ {
		var j, minIndex = i, i

		for ; j < len(arr)-1; j++ {

			if arr[minIndex] > arr[j+1] {
				minIndex = j + 1
			}

		}

		util.Swap(arr, minIndex, i)
	}

}

File: py/sort/select_sort.py
from sort import *


def select_sort(arr):
    i = 0
    length = len(arr)
    while i < length:
        j = i
        minIndex = i
        while j < length - 1:
            if arr[minIndex] > arr[j + 1]:
                minIndex = j + 1
            j = self_incr(j)

        swap(arr, i, minIndex)

        i = self_incr(i)


if __name__ == '__main__':
    arr = [12, 34, 11, 35, 54, 1, 5, 78, 3]

    select_sort(arr)

    # print(dir(_thread))
    print(arr)
